bitnami/opencart/templates/_helpers.tpl

```diff
@@ -53,135 +21,110 @@ Note, returns 127.0.0.1 if using ClusterIP.
 {{/*
 Gets the host to be used for this application.
 If not using ClusterIP, or if a host or LoadBalancerIP is not defined, the value will be empty.
+When using Ingress, it will be set to the Ingress hostname.
 */}}
 {{- define "opencart.host" -}}
+{{- if .Values.ingress.enabled }}
+{{- $host := (index .Values.ingress.hosts 0).name | default "" -}}
```

Comment:
I think this could be problematic, this `index .Values.ingress.hosts 0` force `hosts` to have at least 1 element when `ingress.enabled=true` otherwise it will fail. Although we are offering `ingress.hostname, it could not be used since  `hosts` needs to have at least 1 element then:

- if you use `ingress.enabled=true,ingress.hostsname=myhost.com` this helper will fail
- if you use `ingress.enabled=true,ingress.hosts[0].name=myhost.com` this helper will not fail but you will have an entry in your ingress which hosts will be empty.
- if you use `ingress.enabled=true,ingress.hostname=myhost.com,,ingress.hosts[0].name=myhost.com` this helper will not fail but will have a duplicate entry in your ingress.

Reply:
It seems like you're right, good catch! I'm changing this